ports/esp32/Makefile

```diff
@@ -89,7 +91,7 @@ else ifeq ($(ESPIDF_CURHASH),$(ESPIDF_SUPHASH_V4))
 $(info Building with ESP IDF v4)
 
 PYPARSING_VERSION = $(shell python3 -c 'import pyparsing; print(pyparsing.__version__)')
-ifneq ($(PYPARSING_VERSION),2.3.1)
+ifeq ($(filter 2.3% 2.2% 2.1%,$(PYPARSING_VERSION)),)
```

Comment:
Is it really a good idea to allow all these older versions? Did you test against each one? Do we really want to test against each one?